ui/components/DataTable.tsx

```diff
@@ -453,7 +453,7 @@ function UnstyledDataTable({
 
   const r = _.map(sorted, (r, i) => {
     return (
-      <TableRow key={r.uid || i}>
```

Comment:
Yes - SOMEHOW these uids were appearing twice in the page and it was throwing an error

Reply:
Maybe it's just an indicator of an issue with the payload (have not found where exactly, looks OK to me in general) or listing objects when testing.

- This is what I get as normal object data when loading the Automations page with  tilt up:

<img width="856" alt="Screenshot 2022-10-26 at 22 02 15" src="https://user-images.githubusercontent.com/8824708/198127156-b69eb29b-a16f-47f0-b00b-73588c62609a.png">

The UIDs look unique.

- This is what I get when running the test and adding printing automations to the console in `Automations.tsx`:

```
console.log
    automations:

      at Automations (ui/pages/v2/Automations.tsx:14:11)

  console.log
    {
      result: [
        HelmRelease {
          obj: [Object],
          clusterName: 'Default',
          tenant: '',
          uid: '907093ec-5471-46f9-9953-b5b36f9f7859',
          inventory: []
        },
        HelmRelease {
          obj: [Object],
          clusterName: 'Default',
          tenant: '',
          uid: '2dd24865-4ae4-4a0e-9c78-3204a470be9f',
          inventory: []
        },
        Kustomization {
          obj: [Object],
          clusterName: 'Default',
          tenant: '',
          uid: '907093ec-5471-46f9-9953-b5b36f9f7859'
        },
        Kustomization {
          obj: [Object],
          clusterName: 'Default',
          tenant: '',
          uid: '2dd24865-4ae4-4a0e-9c78-3204a470be9f'
        }
      ],
      errors: []
    }

      at Automations (ui/pages/v2/Automations.tsx:15:11)
```

So, there are two pairs of objects with duplicate UIDs, one in each pair is a HelmRelease and one a Kustomization.

The table is doing what is expected. I think we can keep assigning the UIDs to rows as keys (because AFAIK the normal behavior is that all object UIDs in the response are unique) and fix the duplicate objects instead?

Unfortunately, I don't know this code well enough to say why exactly there are duplicates.

Reply:
Fixed - the way I wrote the mock response caused each object to be returned twice, since useListAutomations runs ListObjects for both Kustomizations and HelmReleases - I supplied the req as an argument to the mock and used kind, and that fixed it! Thanks for looking into it